Essentials/src/com/earth2me/essentials/Worth.java

```diff
@@ -62,6 +62,86 @@ public BigDecimal getPrice(IEssentials ess, ItemStack itemStack) {
         return result;
     }
 
+    /**
+     * Get the value of an item stack from the config for buying.
+     *
+     * @param ess       The Essentials instance.
+     * @param itemStack The item stack to look up in the config.
+     * @return The price from the config.
+     */
+    public BigDecimal getBuyPrice(IEssentials ess, ItemStack itemStack) {
+        BigDecimal result;
+
+        String itemname = itemStack.getType().toString().toLowerCase(Locale.ENGLISH).replace("_", "");
+
+        // Check for matches with data value from stack
+        // Note that we always default to BigDecimal.ONE.negate(), equivalent to -1
+        result = config.getBigDecimal("buy." + itemname + "." + itemStack.getDurability(), BigDecimal.ONE.negate());
```

Comment:
nit: Don't separate the declaration from the initial assignment - there isn't any reason to separate it.

Reply:
This was fixed in my refactor.